### PR TITLE
Add setting for throwing an exception when filter is not defined

### PR DESCRIPTION
**Pull request by Tomas2D**
Based on issue #11

### CODE REVIEW

File: QueryFilter/QueryFilter.php [hunk 1]
```diff
@@ -124,11 +126,14 @@ private function getSimpleSearchBy(array $allowedCols, ?array $search): array
         }
 
         foreach ($search as $key => $val) {
-            if (!in_array($key, $allowedCols, true) || $val === null) {
+            if (in_array($key, $allowedCols, true) && $val !== null) {
```
Comment (denisvmedia) [inline QueryFilter/QueryFilter.php:143]: Thank you. I just noticed a small inconsistency of the approach. Here you inverted the condition and moved the `throw` call below. But in `getFullSearchBy`. You just added an additional condition inside the existing condition. I don't mind against any of the approaches, but it's better to have it consistent.

Also, did you think of the sort condition? Do we want to have it handled by the new feature as well?

File: QueryFilter/QueryFilter.php [hunk 2]
```diff
@@ -201,8 +211,8 @@ private function getSearchBy(ConfigInterface $config): array
     {
         // Get basic search by
         $searchBy = $config->getRequest()->isSimple()
-            ? $this->getSimpleSearchBy($config->getSearchAllowedCols(), $config->getRequest()->getQuery())
-            : $this->getFullSearchBy($config->getSearchAllowedCols(), $config->getRequest()->getQuery());
+            ? $this->getSimpleSearchBy($config->getSearchAllowedCols(), $config->getRequest()->getQuery(), $config->isStrictColumns())
```
Comment (denisvmedia) [inline QueryFilter/QueryFilter.php:227]: Would you mind to add a couple of tests to QueryFilterTest.php to cover the new option?

File: QueryFilter/QueryFilter.php (hunk 1 above)
Comment (Tomas2D) [reply QueryFilter/QueryFilter.php:143]: I rewrote the check in `getFullSearchBy` method to match. Is it okay right now? I personally do not prefer having nested if statements if it is not necessarily.

Comment (Tomas2D) [reply QueryFilter/QueryFilter.php:143]: Good point. I also added check for column/type in `getSortData` method in `QueryFilter` class.

File: QueryFilter/QueryFilter.php (hunk 2 above)
Comment (Tomas2D) [reply QueryFilter/QueryFilter.php:227]: Sure, test cases added.

File: QueryFilter/QueryFilter.php [hunk 3]
```diff
@@ -73,14 +74,28 @@ private function getSortData(ConfigInterface $config): array
             'field' => $config->getRequest()->getSortBy(),
             'type' => $config->getRequest()->getSortDir(),
         ];
-        $sortData = $config->getSortColsDefault();
-        if (isset($sort['field'], $sort['type'])) {
-            if (in_array($sort['type'], array('asc', 'desc'), true) && in_array($sort['field'], $config->getSortCols(), true)) {
-                $sortData = array($sort['field'] => $sort['type']);
+
+        if (!isset($sort['field'], $sort['type'])) {
```
Comment (denisvmedia) [inline QueryFilter/QueryFilter.php:78]: Good job!

File: README.md [hunk 4]
```diff
@@ -54,6 +54,9 @@ class DefaultController extends Controller
         $config->setSortCols(['t.id'], ['t.id' => 'asc']);
         $config->setRequest(new Request($request));
         
+        // Throws an UnexpectedValueException when invalid filter column is specified
```
Comment (denisvmedia) [inline README.md]: Just to make it to better reflect the real behavior, I suggest to slightly extend the comment to the following one:

> Throws an UnexpectedValueException when invalid filter column, sort column or sort type is specified

File: Tests/QueryFilter/QueryFilterTest.php [hunk 5]
```diff
@@ -92,6 +92,49 @@ public function testGetDataSimpleBaseCase()
...
+
+        $response = $queryFilter->getData($config);
+        $response->getData();
+    }
+
+    /**
+     * @test
+     * @expectedException \Artprima\QueryFilterBundle\Exception\UnexpectedValueException
+     */
+    public function testGetDataInvalidSortColumn()
```
Comment (denisvmedia) [inline Tests/QueryFilter/QueryFilterTest.php:121]: Thank you, looks good to me. Would you please also add a test for invalid sort type?

File: QueryFilter/QueryFilter.php (hunk 1 above)
Comment (denisvmedia) [reply QueryFilter/QueryFilter.php:143]: I liked your changes. Thank you! There are a few minor issues left, and we are done. 👍

**Comment by denisvmedia** [comment]
All in all it looks good, one test is just missing, and the feature is ready 👍

File: Tests/QueryFilter/QueryFilterTest.php (hunk 5 above)
Comment (Tomas2D) [reply Tests/QueryFilter/QueryFilterTest.php:121]: I wanted to, but you already did that. Look at `RequestTest.php` file to method called `testInvalidSortDirException1` and `testInvalidSortDirException2`

Comment (denisvmedia) [reply Tests/QueryFilter/QueryFilterTest.php:121]: I see, ok, let's keep it as it is.

**Review by denisvmedia — APPROVED** [review]

File: README.md (hunk 4 above)
Comment (Tomas2D) [reply README.md]: Good point, edited.

**Pull request merged**